vcd/resource_vcd_nsxt_edgegateway_bgp_neighbor_test.go

```diff
@@ -0,0 +1,264 @@
+				),
+			},
+		},
+	})
+}
+
+const testAccVcdNsxtBgpNeighborConfigVdcGroupPrereqs = `
+data "vcd_vdc_group" "g1" {
+  org  = "{{.Org}}"
+  name = "{{.NsxtVdcGroup}}"
+}
+
+data "vcd_nsxt_edgegateway" "testing" {
+  org      = "{{.Org}}"
+  owner_id = data.vcd_vdc_group.g1.id
+
+  name = "{{.NsxtEdgeGwInVdcGroup}}"
+}
+`
+
+const testAccVcdNsxtBgpNeighborVdcGroupConfig1 = testAccVcdNsxtBgpNeighborConfigVdcGroupPrereqs + `
+resource "vcd_nsxt_edgegateway_bgp_neighbor" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  ip_address       = "1.1.1.1"
+  remote_as_number = "65211"
+  
+  password              = "phinai0ca,iS"
+  keep_alive_timer      = 80
+  hold_down_timer       = 321
+  graceful_restart_mode = "HELPER_ONLY"
+  allow_as_in           = true
+  bfd_enabled           = true
+  route_filtering       = "DISABLED"
+  
+}
+`
+
+const testAccVcdNsxtBgpNeighborVdcGroupConfig2DS = testAccVcdNsxtBgpNeighborVdcGroupConfig1 + `
+data "vcd_nsxt_edgegateway_bgp_neighbor" "testing" {
+  org = "{{.Org}}"
+  
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  ip_address = vcd_nsxt_edgegateway_bgp_neighbor.testing.ip_address
+
+  depends_on = [vcd_nsxt_edgegateway_bgp_neighbor.testing]
+}
+`
+
+const testAccVcdNsxtBgpNeighborVdcGroupConfig3 = testAccVcdNsxtBgpNeighborConfigVdcGroupPrereqs + `
+resource "vcd_nsxt_edgegateway_bgp_neighbor" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  ip_address       = "1.1.1.1"
+  remote_as_number = "62513"
+
+  keep_alive_timer      = 78
+  hold_down_timer       = 235
+  graceful_restart_mode = "HELPER_ONLY"
+  allow_as_in           = true
+  bfd_enabled           = true
+  bfd_interval          = 800
+  bfd_dead_multiple     = 5
+  route_filtering       = "DISABLED"
+}
+`
+
+const testAccVcdNsxtBgpNeighborVdcGroupConfig4DS = testAccVcdNsxtBgpNeighborVdcGroupConfig3 + `
+data "vcd_nsxt_edgegateway_bgp_neighbor" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  ip_address = vcd_nsxt_edgegateway_bgp_neighbor.testing.ip_address
+  
+
+  depends_on = [vcd_nsxt_edgegateway_bgp_neighbor.testing]
+}
+`
+
+const testAccVcdNsxtBgpNeighborVdcGroupConfig6 = testAccVcdNsxtBgpNeighborConfigVdcGroupPrereqs + `
+resource "vcd_nsxt_edgegateway_bgp_neighbor" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  ip_address       = "1.1.1.1"
+  remote_as_number = "62513"
+
+  keep_alive_timer      = 78
+  hold_down_timer       = 400
+  graceful_restart_mode = "GRACEFUL_AND_HELPER"
+  allow_as_in           = false
+  bfd_enabled           = false
+  bfd_interval          = 800
+  bfd_dead_multiple     = 5
+  route_filtering       = "IPV4"
+  
+  # TODO - integrate tests when BGP IP Prefix List PRs are merged
```

Comment:
Just reminding that this is here :)

Reply:
Ok. This will happen in a separate PR once I have it all in `main`